xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml

```diff
@@ -701,9 +704,15 @@
  * NOTE: This macro uses variables defined in livetable_filterProperty . It was not meant to be used alone.
  *#
 #macro (livetable_filterDateProperty)
-  ## TODO: The date filter should be smarter than this..
-  #set ($whereSql = "${whereSql} and upper(str(${safe_tableAlias}.value)) like upper(?)")
-  #set ($discard = $whereParams.add("%$filterValue%"))
+  #if ($safe_tableAlias.startsWith('doc.'))
+    #set ($dateProperty = $safe_tableAlias)
+  #else
+    #set ($dateProperty = "${safe_tableAlias}.value")
+  #end
+  #set ($timestamps = $filterValue.split('-'))
+  #set ($whereSql = "${whereSql} and $dateProperty >= ? and $dateProperty <= ?")
+  #set ($discard = $whereParams.add($datetool.toDate($numbertool.toNumber($timestamps[0]))))
+  #set ($discard = $whereParams.add($datetool.toDate($numbertool.toNumber($timestamps[1]))))
```

Comment:
Technically, no, but the daterangepicker control I'm using right now doesn't support that - not sure how to go about this without adding to the control. A checkbox to signify open-ended?

As for the bulletproofing, just a simple `$filterValue.contains('-')`, `$timestamps.count == 2` and `$numbertoo.isNumber($timestamp[0])` should do!